dotnet/src/SemanticKernel.Abstractions/AI/ChatCompletion/ChatCompletionExtensions.cs

```diff
@@ -30,10 +30,37 @@ await foreach (var chatMessageStream in chatCompletionResult.GetStreamingChatMes
             {
                 yield return chatMessageStream.Content;
             }
+
             yield break;
         }
     }
 
+    /// <summary>
+    /// Generate a new chat message
+    /// </summary>
+    /// <param name="chatCompletion">Target interface to extend</param>
+    /// <param name="chat">Chat history</param>
+    /// <param name="requestSettings">AI request settings</param>
+    /// <param name="cancellationToken">Async cancellation token</param>
+    /// <returns>Stream the generated chat message in string format</returns>
+    public static async IAsyncEnumerable<string> GenerateMessagesStreamAsync(
```

Comment:
Remove this extension, the first will work for your scenarios as the `WithData` doesn't support multiple Results / Choices.

This method will also Bug/Freeze out if used with the current Azure SDK Completions implementation.

Reply:
Thanks! First extension doesn't work in my scenario, so I decided to use `IChatStreamingResult` directly.